sound/soc/sof/pcm.c

```diff
@@ -201,6 +201,14 @@ static int sof_pcm_hw_free(struct snd_pcm_substream *substream)
 	/* send IPC to the DSP */
 	ret = sof_ipc_tx_message(sdev->ipc, stream.hdr.cmd, &stream,
 				 sizeof(stream), &reply, sizeof(reply));
+	/* force power down if FW is in error state */
```

Comment:
@kv2019i 
I suggest to add some debug option here to give developer some chance to recover the scene. As we will lost the error or the panic exception in memory bank as the DSP is reset.

Reply:
@xiulipan thanks for the review! 

This is actually not so bad for debug as the client will be stuck waiting on sample availability. So e.g. with aplay, you have plenty of time to dump the DSP memories, before you hit ctrl-C or otherwise kill the application. 

There may of course be issue with apps that have their own timeout mechanism and will force the app to close if the audio device seems stuck. 

This seems like a very specific debug option -- do you think it's worth its own Kconfig option or is it enough just to have a separate function that can be commented out easily by developers when debugging?